include/pagmo/utils/generic.hpp

```diff
@@ -310,6 +310,100 @@ inline vector_double random_decision_vector(const problem &prob, Rng &r_engine)
     return out;
 }
 
+/// Generate a batch of random decision vectors compatible with a problem
+/**
+ * This function will generate \p n decision vectors whose values
+ * are randomly chosen with uniform probability within
+ * the input problem's bounds. The decision vectors are laid
+ * out contiguously in the return value: for a problem with dimension \f$ x \f$,
+ * the first decision vector in the return value occupies
+ * the index range \f$ \left[0, x\right) \f$, the second decision vector occupies the range
+ * \f$ \left[x, 2x\right) \f$, and so on.
+ *
+ * For the continuous parts of the decision vectors, the values will be
+ * generated via pagmo::uniform_real_from_range().
+ *
+ * For the discrete parts of the decision vectors, the values will be generated
+ * via pagmo::uniform_integral_from_range().
+ *
+ * @param prob the input pagmo::problem
+ * @param n how many decision vectors will be generated
+ * @param r_engine a C++ random engine
+ *
+ * @throws std::overflow_error in case of (unlikely) overflow errors.
+ * @throws unspecified any exception thrown by pagmo::uniform_real_from_range() or pagmo::uniform_integral_from_range().
+ *
+ * @returns a pagmo::vector_double containing \p n random decision vectors
+ */
+template <typename Rng>
+inline vector_double batch_random_decision_vector(const problem &prob, vector_double::size_type n, Rng &r_engine)
+{
+    // NOTE: it is possible in principle to do this in a parallel fashion, e.g., see code
+    // at the commit 13d4182a41e4e71034c6e1085699c5138805d21c. The price to pay however is
+    // the loss of determinism. We can reconsider in the future whether it's worth it
+    // to add an option to this constructor, e.g., par_random, defaulting to false.
+
+    // Fetch a few quantities from prob.
+    const auto nx = prob.get_nx();
+    const auto ncx = nx - prob.get_nix();
+    const auto &lb = prob.get_lb();
+    const auto &ub = prob.get_ub();
+
+    // LCOV_EXCL_START
+    if (n > std::numeric_limits<vector_double::size_type>::max() / nx) {
+        pagmo_throw(std::overflow_error,
+                    "Cannot generate " + std::to_string(n)
+                        + " random decision vectors in batch mode, as that would result in an overflow error");
+    }
+    // LCOV_EXCL_STOP
+
+    // Check the problem bounds, and prepare the random distributions.
+    std::vector<std::uniform_real_distribution<double>> v_rdist;
```

Comment:
It is weird some how to be forced to create these large vectors of distribution. Would it not be somehow more efficient to use the distribution parameter setters (https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution/param, https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/param)

then you only need to construct one uniform_int and one uniform_real distribution and change the param on the fly.